components/sup/src/manager/service/mod.rs

```diff
@@ -447,82 +407,44 @@ impl Service {
                 None
             }
         };
-        let mut rumor = ServiceRumor::new(self.local_member_id.clone(),
-                                          &self.package().ident,
+        let mut rumor = ServiceRumor::new(self.sys.member_id.as_str(),
+                                          &self.pkg.ident,
                                           &self.service_group,
-                                          &*self.config.sys,
+                                          &self.sys.as_sys_info(),
                                           exported.as_ref());
         rumor.set_incarnation(incarnation);
         rumor
     }
 
     /// Run initialization hook if present
-    fn initialize(&mut self) {
+    fn initialize(&mut self, census_ring: &CensusRing) {
         if self.initialized {
             return;
         }
         outputln!(preamble self.service_group, "Initializing");
-        self.hooks.compile(&self.service_group, &self.config);
-        if let Some(err) = self.copy_run().err() {
-            outputln!(preamble self.service_group, "Failed to copy run hook: {}", err);
+        {
+            let ctx = self.render_context(&census_ring);
```

Comment:
If this can be avoided and run in the `update_configuration` method I would find that much better.
Perhaps there could be code like:
```
if !self.initialized {
  ... always render everything
   return
}
```
in the `update_configuration` method.